Sources/S5TF/utils/S5TFUtils.swift

```diff
@@ -60,4 +60,79 @@ public struct S5TFUtils {
 
         return (output, task.terminationStatus)
     }
+
+    /// Extract a downloaded archive
+    /// 
+    /// - Parameters:
+    ///   - `fileAtURL`: the URL for the archive.
+    ///
+    /// - Returns: output, termination status of the command.
+    ///
+    /// - Usage Example:
+    ///   - Extract "archive.tgz":
+    ///     ```
+    ///     extract(fileAt: URL(string: "archive.tgz")!)
+    ///     ```
+    ///   - Extract "another_archive.gz"
+    ///     ```
+    ///     extract(fileAt: URL(string: "archive_archive.gz")!)
+    ///     ```
+    ///
+    static public func extract(archiveURL: URL) -> URL {
+        let path = archiveURL.path
+        let fileExtension = archiveURL.pathExtension
+
+        #if os(macOS)
+            let binary = "/usr/bin/"
+        #else
+            let binary = "/bin/"
+        #endif
+
+        let tool: String
+        let arguments: [String]
+
+        switch fileExtension {
+            case "gz":
+                tool = "gunzip"
+                arguments = [path]
+            case "tar.gz", "tgz":
+                tool = "tar"
+                arguments = ["xzf", path]
+            default:
+                fatalError("Unsupported file extension for archive.")
+        }
+        if !FileManager.default.fileExists(atPath: archiveURL.absoluteString.deletingPathExtension) {
+            do {
+                try shell(binary + tool, arguments)
+            } catch {
+                fatalError("Error extracting file.")
+            }
+        }
+
+        return URL(string: archiveURL.absoluteString.deletingPathExtension)!
+    }
+
+    /// Download and extract an archive.
+    /// 
+    /// - Parameters:
+    ///   - `remoteURL`: the remote URL for the archive.
+    ///   - `cacheName`: name of the cache.
+    ///   - `fileName`: name of the file.
+    ///
+    /// - Returns: URL of extracted archive.
+    ///
+    /// - Usage Example:
+    ///   - Download and extract "https://storage.googleapis.com/cvdf-datasets/mnist/train-images-idx3-ubyte.gz":
+    ///     ```
+    ///     downloadAndExtract(remoteURL: URL(string: "https://storage.googleapis.com/cvdf-datasets/mnist/train-images-idx3-ubyte.gz")!,
+    ///                        cacheName: "mnist", fileName: "train_images")
+    ///     ```
+    ///
+    @discardableResult
+    static public func downloadAndExtract(remoteURL: URL, cacheName: String, fileName: String) -> URL? {
+        let archiveURL = Downloader.download(fileAt: remoteURL,
+                                             cacheName: cacheName,
+                                             fileName: fileName)
+        return extract(archiveURL: archiveURL!)
+    }
```

Comment:
I think we are not sure whether `Downloader` will always return a URL. Let's add a little `guard`:

```swift
guard let archiveURL = archiveURL else {
    return nil
}
```